src/vs/code/electron-main/windows.ts

```diff
@@ -441,7 +441,17 @@ export class WindowsManager implements IWindowsMainService {
 		}
 
 		// Handle empty
-		if (emptyToOpen.length > 0) {
+		if (emptyToRestore.length > 0) {
+			emptyToRestore.forEach(backupFolder => {
+				const configuration = this.toConfiguration(openConfig, null, null, null, null, backupFolder);
+				const browserWindow = this.openInBrowserWindow(configuration, openInNewWindow, openInNewWindow ? void 0 : openConfig.windowToUse);
```

Comment:
@Tyriar we always want each of these windows to open inside a new window so I suggest to just call 

`const browserWindow = this.openInBrowserWindow(configuration, true /* new window */);`

In the same way as we do for restoring folders.